network/src/node.rs

```diff
@@ -55,60 +54,34 @@ pub struct InnerNode<S: Storage> {
     pub config: Config,
     /// The inbound handler of this node.
     pub inbound: Inbound,
-    /// The outbound handler of this node.
-    pub outbound: Outbound,
     /// The list of connected and disconnected peers of this node.
     pub peer_book: PeerBook,
     /// The sync handler of this node.
     pub sync: OnceCell<Arc<Sync<S>>>,
     /// The node's start-up timestamp.
     pub launched: DateTime<Utc>,
     /// The tasks spawned by the node.
-    tasks: Mutex<Vec<task::JoinHandle<()>>>,
+    tasks: DropJoin<task::JoinHandle<()>>,
     /// The threads spawned by the node.
-    threads: Mutex<Vec<thread::JoinHandle<()>>>,
+    threads: DropJoin<thread::JoinHandle<()>>,
     /// An indicator of whether the node is shutting down.
     shutting_down: AtomicBool,
 }
 
-impl<S: Storage> Drop for InnerNode<S> {
```

Comment:
why is this `Drop` impl removed? I assume it's because the strong reference count to `InnerNode` has increased and there is no chance for it to be triggered? this might have been the case for some time now

Reply:
`connected_peers` no longer needs aborting due to channel drop = close. i think i discounted the other two, but yeah this is unreachable anyways, even for tests.